src/views/DidUpgrade/DidUpgrade.tsx

```diff
@@ -0,0 +1,121 @@
+import { browser } from 'webextension-polyfill-ts';
+
+import { Link } from 'react-router-dom';
+
+import { FormEvent, useCallback } from 'react';
+
+import * as styles from './DidUpgrade.module.css';
+
+import { Identity } from '../../utilities/identities/types';
+import { IdentitySlide } from '../../components/IdentitySlide/IdentitySlide';
+import { paths } from '../paths';
+import {
+  PasswordField,
+  usePasswordField,
+} from '../../components/PasswordField/PasswordField';
+import { LinkBack } from '../../components/LinkBack/LinkBack';
+import { getTransaction } from '../../utilities/didUpgrade/didUpgrade';
+
+import { useAsyncValue } from '../../utilities/useAsyncValue/useAsyncValue';
+import { useTXDSubmitter } from '../../utilities/useTXDSubmitter/useTXDSubmitter';
+import {
+  getIdentityCryptoFromSeed,
+  getLightDidFromSeed,
+} from '../../utilities/identities/identities';
+import {
+  getCheckoutURL,
+  getCheckoutCosts,
+} from '../../utilities/checkout/checkout';
+
+interface Props {
+  identity: Identity;
+}
+
+export function DidUpgrade({ identity }: Props): JSX.Element | null {
+  const t = browser.i18n.getMessage;
+
+  const { address } = identity;
+
+  const passwordField = usePasswordField();
+
+  const cost = useAsyncValue(getCheckoutCosts, [])?.did;
+  const submitter = useTXDSubmitter();
+
+  const handleSubmit = useCallback(
+    async (event: FormEvent) => {
+      event.preventDefault();
+
+      if (!submitter) {
+        throw new Error('Submitter address missing');
+      }
+
+      const { seed } = await passwordField.get(event);
+      const { keypair, sign } = await getIdentityCryptoFromSeed(seed);
+
+      const document = getLightDidFromSeed(seed);
+      const { extrinsic } = await getTransaction(
+        document,
+        keypair,
+        sign,
+        submitter,
+      );
+
+      const checkout = await getCheckoutURL();
+
+      const url = new URL(checkout);
+      url.searchParams.set('address', address);
```

Comment:
While the checkout does not support it yet, we want to have it in the future, so we can use this moment to include `url.searchParams.set('did', fullDid);` here